examples/teleport-tokens/BasicERC20.t.sol

```diff
@@ -60,29 +60,19 @@ contract GmpTestToolsTest is Test {
         assertEq(address(shibuyaErc20), vm.computeCreateAddress(ALICE, 0), "unexpected shibuyaErc20 address");
         assertEq(address(sepoliaErc20), vm.computeCreateAddress(BOB, 0), "unexpected sepoliaErc20 address");
 
-        ///////////////////////////////////////////////////////////
-        // Step 3: Deposit funds to destination Gateway Contract //
-        ///////////////////////////////////////////////////////////
-
-        // Switch to Sepolia network and Alice account
-        GmpTestTools.switchNetwork(SEPOLIA_NETWORK, ALICE);
-        // If the sender is a contract, it's address must be converted
-        GmpSender sender = address(shibuyaErc20).toSender(true);
-        // Alice deposit 1 ether to Sepolia gateway contract
-        SEPOLIA_GATEWAY.deposit{value: 1 ether}(sender, SHIBUYA_NETWORK);
-
         //////////////////////////////
-        // Step 4: Send GMP message //
+        // Step 3: Send GMP message //
         //////////////////////////////
 
         // Switch to Shibuya network and Alice account
         GmpTestTools.switchNetwork(SHIBUYA_NETWORK, ALICE);
 
-        // Teleport 100 tokens from Alice to to Bob's account in sepolia
-        // Obs: The `teleport` method internally calls `gateway.submitMessage(...)`
+        // Teleport 100 tokens from Alice to Bob's account in Sepolia
+        // Obs: The `teleport` method calls `gateway.submitMessage(...)` with value
+        uint256 deposit = 2400000000000000000;
```

Comment:
The deposit must be estimated, not hard coded

Reply:
In this case specifically, you must add a new method in the `BasicERC20` for estimate the gas cost, because only this contract knows how the message is formated, and it's size.

Reply:
updated BasicERC20.t  and added teleportCost function in BasicERC20.sol